notebooks/integrations/cohere/cohere-elasticsearch.ipynb

```diff
@@ -0,0 +1,562 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# https://www.elastic.co/search-labs/tutorials/install-elasticsearch/elastic-cloud#finding-your-cloud-id\n",
+    "ELASTIC_CLOUD_ID = getpass(\"Elastic Cloud ID: \")\n",
+    "\n",
+    "# https://www.elastic.co/search-labs/tutorials/install-elasticsearch/elastic-cloud#creating-an-api-key\n",
+    "ELASTIC_API_KEY = getpass(\"Elastic API key: \")\n",
+    "\n",
+    "# Create the client instance\n",
+    "client = Elasticsearch(\n",
+    "    # For local development\n",
+    "    # hosts=[\"http://localhost:9200\"]\n",
+    "    cloud_id=ELASTIC_CLOUD_ID,\n",
+    "    api_key=ELASTIC_API_KEY,\n",
+    ")\n",
+    "\n",
+    "# Confirm the client has connected\n",
+    "print(client.info())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3d29b809",
+   "metadata": {},
+   "source": [
+    "## Create the inference endpoint\n",
+    "\n",
+    "Create the inference endpoint first. In this example, the inference endpoint \n",
+    "uses Cohere's `embed-english-v3.0` model and the `embedding_type` is set to\n",
+    "`byte`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ae8ae88c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "COHERE_API_KEY = getpass(\"Cohere API key: \")\n",
+    "co = cohere.Client(api_key=COHERE_API_KEY)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6e524ce4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "client.inference.put_model(\n",
+    "    task_type=\"text_embedding\",\n",
+    "    inference_id=\"cohere_embeddings1\",\n",
+    "    body={\n",
+    "        \"service\": \"cohere\",\n",
+    "        \"service_settings\": {\n",
+    "            \"api_key\": COHERE_API_KEY,\n",
+    "            \"model_id\": \"embed-english-v3.0\",\n",
+    "            \"embedding_type\": \"byte\",\n",
+    "        },\n",
+    "    },\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "884e424d",
+   "metadata": {},
+   "source": [
+    "You can find your API keys in your Cohere dashboard under the\n",
+    "[API keys section](https://dashboard.cohere.com/api-keys)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "178c32db",
+   "metadata": {},
+   "source": [
+    "## Create the index mapping\n",
+    "\n",
+    "Create the index mapping for the index that will contain the embeddings."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "35ab26b2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "client.indices.create(\n",
+    "    index=\"cohere-embeddings\",\n",
+    "    settings={\"index\": {\"default_pipeline\": \"cohere_embeddings\"}},\n",
+    "    mappings={\n",
+    "        \"properties\": {\n",
+    "            \"text_embedding\": {\n",
+    "                \"type\": \"dense_vector\",\n",
+    "                \"dims\": 1024,\n",
+    "                \"element_type\": \"byte\",\n",
```

Comment:
The `element_type` key is not in the spec. Is this something new that needs to be added?

Reply:
I don't think it's new. It's a [parameter of the `dense_vector` field type](https://www.elastic.co/guide/en/elasticsearch/reference/current/dense-vector.html#dense-vector-params) that is available since 8.6.

Reply:
Oh, okay, this is strange. I see it in the [typescript definition](https://github.com/elastic/elasticsearch-specification/blob/main/specification/_types/mapping/complex.ts#L52-L59), but for some reason it does not appear in the generated JSON spec. Need to figure out why.

Reply:
Awesome, thank you for the thorough investigation @miguelgrinberg!